pallets/order-book/src/types.rs

```diff
@@ -186,11 +224,169 @@ impl<AssetId: PartialEq> DealInfo<AssetId> {
 }
 
 #[derive(Eq, PartialEq, Clone, RuntimeDebug)]
-pub struct MarketChange<AccountId, OrderId, LimitOrder, BlockNumber> {
-    pub market_input: OrderAmount,
-    pub market_output: OrderAmount,
+pub struct Payment<AssetId, AccountId, DEXId> {
+    pub dex_id: DEXId,
+    pub order_book_id: OrderBookId<AssetId>,
+    pub to_lock: BTreeMap<AssetId, BTreeMap<AccountId, OrderVolume>>,
+    pub to_unlock: BTreeMap<AssetId, BTreeMap<AccountId, OrderVolume>>,
+}
+
+impl<AssetId, AccountId, DEXId> Payment<AssetId, AccountId, DEXId>
+where
+    AssetId: Copy + PartialEq + Ord,
+    AccountId: Ord + Clone,
+    DEXId: Copy + PartialEq,
+{
+    pub fn new(dex_id: DEXId, order_book_id: OrderBookId<AssetId>) -> Self {
+        Self {
+            dex_id,
+            order_book_id,
+            to_lock: BTreeMap::new(),
+            to_unlock: BTreeMap::new(),
+        }
+    }
+
+    pub fn merge(&mut self, other: &Self) -> Result<(), ()> {
+        ensure!(self.dex_id == other.dex_id, ());
+        ensure!(self.order_book_id == other.order_book_id, ());
+
+        let merge_asset_map =
+            |map: &mut BTreeMap<AssetId, BTreeMap<AccountId, OrderVolume>>,
+             to_merge: &BTreeMap<AssetId, BTreeMap<AccountId, OrderVolume>>| {
+                to_merge.iter().for_each(|(asset, whom)| {
```

Comment:
Maybe use simple `for`? Should look cleaner, what do you think?

Reply:
No difference